app/server/lib/SqliteNode.ts

```diff
@@ -113,6 +113,10 @@ export class NodeSqlite3DatabaseAdapter implements MinDB {
 
   public async limitAttach(maxAttach: number) {
     const SQLITE_LIMIT_ATTACHED = (sqlite3 as any).LIMIT_ATTACHED;
+    // Work around node-sqlite3 bug when `.configure()` is called while a
+    // query is running.
+    // See this issue upstream: https://github.com/TryGhost/node-sqlite3/issues/1838
+    await new Promise(resolve => (this._db as any).wait(resolve));
```

Comment:
Is there any risk of a tiny gap between wait() finishing and us getting to call configure()? Could another query sneak in there? If so would we end up deadlocked again?

Thanks for the research into this and reporting upstream. Good detective work! The repo looks disturbingly quiet this year :(. I'm excited for native sqlite3 support in node...

Reply:
> Is there any risk of a tiny gap between wait() finishing and us getting to call configure()? Could another query sneak in there? If so would we end up deadlocked again?

I have found a condition that might occur. The test failed before the fix, I hope that reflects your concerns.

@jonathanperret suggested a fix that I have added.

I hope to have added a meaningful comment to explain the idea behind and given evidence it should work around correctly.